File: mp3gain2026-net10/AppSettings.cs
using System;
using System.IO;
using System.Text.Json;

namespace Mp3Gain2026;

public class AppSettings
{
    public bool AlwaysOnTop { get; set; } = false;
    public bool AddSubfolders { get; set; } = true;
    public bool PreserveTimestamp { get; set; } = true;
    public bool NoLayerCheck { get; set; } = false;
    public bool DontClip { get; set; } = true;
    public bool IgnoreTags { get; set; } = false;
    public bool RecalcTags { get; set; } = false;
    public bool BeepWhenFinished { get; set; } = false;

    public string ErrorLogPath { get; set; } = @"C:\ProgramData\Mp3Gain2026\error.log";
    public string AnalysisLogPath { get; set; } = @"C:\ProgramData\Mp3Gain2026\analysis.log";
    public string ChangeLogPath { get; set; } = @"C:\ProgramData\Mp3Gain2026\change.log";

    public decimal TargetVolume { get; set; } = 89.0m;

    /// <summary>
    /// 0 = Path/File, 1 = File Only, 2 = Path and File Separately
    /// </summary>
    public int FilenameDisplayMode { get; set; } = 0;

    private static string GetConfigPath()
    {
        var appData = Environment.GetFolderPath(Environment.SpecialFolder.ApplicationData);
        var dir = Path.Combine(appData, "Mp3Gain2026");
        if (!Directory.Exists(dir))
            Directory.CreateDirectory(dir);
        return Path.Combine(dir, "settings.json");
    }

    public static AppSettings Load()
    {
        var path = GetConfigPath();
        if (File.Exists(path))
        {
            try
            {
                var json = File.ReadAllText(path);
                var settings = JsonSerializer.Deserialize<AppSettings>(json);
                if (settings != null)
                    return settings;
            }
            catch { }
        }
        return new AppSettings();
    }

    public void Save()
    {
        try
        {
            var json = JsonSerializer.Serialize(this, new JsonSerializerOptions { WriteIndented = true });
            File.WriteAllText(GetConfigPath(), json);
        }
        catch { }
    }
}

File: mp3gain2026-net48/mp3gain2026/AppSettings.vb
Imports System.IO
Imports System.Xml.Serialization

Public Class AppSettings

    Public Property AlwaysOnTop As Boolean = False
    Public Property AddSubfolders As Boolean = True
    Public Property PreserveTimestamp As Boolean = True
    Public Property NoLayerCheck As Boolean = False
    Public Property DontClip As Boolean = True
    Public Property IgnoreTags As Boolean = False
    Public Property RecalcTags As Boolean = False
    Public Property BeepWhenFinished As Boolean = False

    Public Property ErrorLogPath As String = "C:\ProgramData\Mp3Gain2026\error.log"
    Public Property AnalysisLogPath As String = "C:\ProgramData\Mp3Gain2026\analysis.log"
    Public Property ChangeLogPath As String = "C:\ProgramData\Mp3Gain2026\change.log"

    Public Property TargetVolume As Decimal = 89.0D

    ''' <summary>
    ''' 0 = Path/File, 1 = File Only, 2 = Path and File Separately
    ''' </summary>
    Public Property FilenameDisplayMode As Integer = 0

    Private Shared Function GetConfigPath() As String
        Dim appData = Environment.GetFolderPath(Environment.SpecialFolder.ApplicationData)
        Dim dir = Path.Combine(appData, "Mp3Gain2026")
        If Not System.IO.Directory.Exists(dir) Then
            System.IO.Directory.CreateDirectory(dir)
        End If
        Return Path.Combine(dir, "settings.xml")
    End Function

    Public Shared Function Load() As AppSettings
        Dim path = GetConfigPath()
        If File.Exists(path) Then
            Try
                Dim serializer As New XmlSerializer(GetType(AppSettings))
                Using reader As New StreamReader(path)
                    Dim settings = TryCast(serializer.Deserialize(reader), AppSettings)
                    If settings IsNot Nothing Then Return settings
                End Using
            Catch
            End Try
        End If
        Return New AppSettings()
    End Function

    Public Sub Save()
        Try
            Dim serializer As New XmlSerializer(GetType(AppSettings))
            Using writer As New StreamWriter(GetConfigPath())
                serializer.Serialize(writer, Me)
            End Using
        Catch
        End Try
    End Sub

End Class
